### PR TITLE
Force Password Reset in LDAP/AD Teams

### PR DESCRIPTION
**Pull request by mhofert-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 17.009ms

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder
workflow1

#### Teams message

teams1

#### Artifact

artifact1

### CODE REVIEW

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/Force_Password_Reset_in_Active_Directory_with_Teams.icon [hunk 1]
```diff
@@ -0,0 +1,1379 @@
+{
+  "kom": {
+    "komandVersion": "1.49.0-0-gc013d",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-04-20T20:07:20.078578904Z",
+    "workflowVersions": [
+      {
+        "name": "Force Password Reset in Active Directory with Teams",
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/Force_Password_Reset_in_Active_Directory_with_Teams.icon:8]:
```suggestion
        "name": "Force Password Reset in Active Directory with Microsoft Teams",
```

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Description
+
+Force an Active Directory user password reset from Microsoft Teams.
+
+Sample Slack command:
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md:5]: Cmd+F: "Slack" 😄

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
+* Microsoft Active Directory
+
+# Documentation
+
+## Setup
+
+Once the workflow has been downloaded, login to InsightConnect and "Import" it into the workflow builder.
+After import, you will initially be prompted to configure the connection for Microsoft Teams and Microsoft Active Directory.
+You will then need to configure the Teams trigger and actions.
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md:24]:
```suggestion
You will then need to configure the Microsoft Teams trigger and actions.
```

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+In the Team and chanel inputs replace `change_me` with the appropriate team and channel.
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md:25]:
```suggestion
In the "Team" and "Channel" inputs replace `change_me` with the appropriate team and channel.
```

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+To run the workflow, in the configured channel and enter the command `!reset_password <username>`.
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/help.md:27]:
```suggestion
To run the workflow, send a message of `!reset_password <username>` in the configured channel.
```

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/workflow.spec.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+extension: workflow
+products: ["insightconnect"]
+name: Force_Password_Reset_in_Active_Directory_with_Teams
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/workflow.spec.yaml:3]:
```suggestion
name: Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams
```
Make sure to rename the workflow directory after making this change

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/workflow.spec.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+title: "Force Password Reset in Active Directory with Teams"
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/workflow.spec.yaml:4]:
```suggestion
title: "Force Password Reset in Active Directory with Microsoft Teams"
```

File: workflows/Force_Password_Reset_in_Active_Directory_with_Teams/workflow.spec.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+description: "Force an Active Directory user password reset from Microsoft Teams."
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [threat_detection_and_response, remediation_management]
+  keywords: [active directory, teams, containment]
```
Comment (jschipp-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [active directory, teams, containment, work_from_home, phishing]
```

File: workflows/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams.icon [hunk 9]
```diff
@@ -0,0 +1,1379 @@
...
+              "slugName": "active_directory_ldap",
+              "slugVersion": "3.2.8",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/active_directory_ldap/3.2.8/icon.png"
+            },
+            "identifier": "query",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "search_base": "DC= Akbar,DC=trap",
```
Comment (mrinehart-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams.icon:246]:
```suggestion
                "search_base": "DC= Akbar,DC=trap",
```
Does this need to be removed?

File: workflows/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [threat_detection_and_response, remediation_management]
+  keywords: [active directory, teams, containment, work_from_home, phishing]
+  features: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Force_Password_Reset_in_Active_Directory_with_Teams
```
Comment (tylerslijboom-r7) [inline workflows/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:15]:
```suggestion
  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams
```

Comment (tylerslijboom-r7) [reply workflows/Force_Password_Reset_in_Active_Directory_with_Microsoft_Teams/workflow.spec.yaml:15]: ...I think?

**Pull request merged**